packages/wrangler/src/pubsub.ts

```diff
@@ -0,0 +1,281 @@
+/**
+ * Delete a Pub/Sub Broker with the given `broker` name within the associated `namespace`.
+ *
+ * Deleting a Broker is destructive and will immediately break all existing
+ * connections to the Broker.
+ */
+export async function deletePubSubBroker(
+  accountId: string,
+  namespace: string,
+  broker: string
+): Promise<void> {
+  return await fetchResult<void>(
+    `/accounts/${accountId}/pubsub/namespaces/${namespace}/brokers/${broker}`,
+    { method: "DELETE" }
+  );
+}
+
+/**
+ * Describe a single Pub/Sub Broker for the given `broker` name within the associated `namespace`.
+ */
+export async function describePubSubBroker(
+  accountId: string,
+  namespace: string,
+  broker: string
+): Promise<void> {
+  return await fetchResult<void>(
+    `/accounts/${accountId}/pubsub/namespaces/${namespace}/brokers/${broker}`,
+    { method: "GET" }
+  );
+}
+
+/**
+ * Fetch a list of all the Pub/Sub Brokers under the given `namespace`.
+ */
+export async function listPubSubBrokers(
+  accountId: string,
+  namespace: string
+): Promise<PubSubBroker[]> {
+  return await fetchResult<PubSubBroker[]>(
+    `/accounts/${accountId}/pubsub/namespaces/${namespace}/brokers`
+  );
+}
+
+/**
+ * Create a Pub/Sub Broker within given `namespace`.
+ */
+export async function createPubSubBroker(
+  accountId: string,
+  namespace: string,
+  broker: PubSubBroker
+): Promise<void> {
+  return await fetchResult<void>(
+    `/accounts/${accountId}/pubsub/namespaces/${namespace}/brokers`,
+    { method: "POST", body: JSON.stringify(broker) }
+  );
+}
+
+/**
+ * Update a Pub/Sub Broker configuration.
+ */
+export async function updatePubSubBroker(
+  accountId: string,
+  namespace: string,
+  broker: string,
+  update: PubSubBrokerUpdate
+): Promise<void> {
+  return await fetchResult<void>(
+    `/accounts/${accountId}/pubsub/namespaces/${namespace}/brokers/${broker}`,
+    { method: "PATCH", body: JSON.stringify(update) }
+  );
+}
+
+/**
+ * Get the public keys uniquely associated with a Pub/Sub Broker.
+ *
+ * These keys can be used for verifying Pub/Sub on-publish hooks over HTTPS.
+ */
+export async function getPubSubBrokerPublicKeys(
+  accountId: string,
+  namespace: string,
+  broker: string
+): Promise<Record<string, string>> {
+  return await fetchResult<Record<string, string>>(
+    `/accounts/${accountId}/pubsub/namespaces/${namespace}/brokers/${broker}/publickeys`
+  );
+}
+
+/**
+ * Issue client credentials for the given `broker`.
+ *
+ * Multiple credentials can be generated at once by providing a `number`.
+ */
+export async function issuePubSubBrokerTokens(
+  accountId: string,
+  namespace: string,
+  broker: string,
+  number: number,
+  type: string,
+  clientIds?: string[],
+  expiration?: number
+): Promise<Record<string, string>> {
+  let url = `/accounts/${accountId}/pubsub/namespaces/${namespace}/brokers/${broker}/credentials?number=${number}&type=${type}`;
+  if (clientIds) {
+    const ids = makeQueryString(clientIds, "clientid");
```

Comment:
This works, but might want to just use URLSearchParams in the future, since it automatically supports proper encoding, multiple values, etc.

Reply:
I'll fix this. The original code relied on the `URL` type, which isn't valid as there's no baseURL set (since it's set at a package level).

The "gap" is that the best you can do is concat `path + "?" + params.toString()`.

Reply:
I've fixed this to use `URLSearchParams` - but we still have to concat the params.

```ts
  let url = `/accounts/${accountId}/pubsub/namespaces/${namespace}/brokers/${broker}/credentials`;

  const params = new URLSearchParams();
  params.append("number", `${number}`);
  params.append("type", type);

  if (clientIds) {
    for (const id of clientIds) {
      params.append("clientid", id);
    }
  }

  if (expiration) {
    params.append("expiration", `${expiration}`);
  }

  // We have to concat these as the URL class requires a base URL, which we
  // don't know within the scope of this function.
  url = url + `?${params.toString()}`;

  return await fetchResult<Record<string, string>>(url);
  ```